pkg/app/piped/executor/terraform/deploy.go

```diff
@@ -152,31 +158,31 @@ func (e *deployExecutor) ensurePlan(ctx context.Context) model.StageStatus {
 	)
 
 	if ok := showUsingVersion(ctx, cmd, e.LogPersister); !ok {
-		return model.StageStatus_STAGE_FAILURE
+		return model.StageStatus_STAGE_FAILURE, provider.PlanResult{}
 	}
 
 	if err := cmd.Init(ctx, e.LogPersister); err != nil {
 		e.LogPersister.Errorf("Failed to init (%v)", err)
-		return model.StageStatus_STAGE_FAILURE
+		return model.StageStatus_STAGE_FAILURE, provider.PlanResult{}
 	}
 
 	if ok := selectWorkspace(ctx, cmd, e.appCfg.Input.Workspace, e.LogPersister); !ok {
-		return model.StageStatus_STAGE_FAILURE
+		return model.StageStatus_STAGE_FAILURE, provider.PlanResult{}
 	}
 
 	planResult, err := cmd.Plan(ctx, e.LogPersister)
 	if err != nil {
 		e.LogPersister.Errorf("Failed to plan (%v)", err)
-		return model.StageStatus_STAGE_FAILURE
+		return model.StageStatus_STAGE_FAILURE, provider.PlanResult{}
 	}
 
 	if planResult.NoChanges() {
 		e.LogPersister.Success("No changes to apply")
-		return model.StageStatus_STAGE_SUCCESS
+		return model.StageStatus_STAGE_SUCCESS, planResult
```

Comment:
How about this without changing the return value?
```suggestion
if e.StageConfig.TerraformPlanStageOptions.ExitOnNoChanges {
        return model.StageStatus_STAGE_EXITING
}
```